purchase_order_line_payment_forecast/models/purchase_order_line.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    price_subtotal_billed = fields.Monetary(
+        compute="_compute_price_subtotal_billed", store=True
+    )
+    price_subtotal_unbilled = fields.Monetary(
+        compute="_compute_price_subtotal_billed", store=True
+    )
+    payment_term_id = fields.Many2one(related="order_id.payment_term_id", store=True)
+    expected_payment_date = fields.Date(
+        compute="_compute_expected_payment_date", store=True
+    )
+
+    @api.depends("price_unit", "qty_invoiced", "qty_to_invoice")
+    def _compute_price_subtotal_billed(self):
+        for line in self:
+            if line.is_deposit:
+                line.price_subtotal_billed = sum(
+                    line.invoice_lines.filtered(
+                        lambda l: l.move_id.state != "cancel"
+                    ).mapped("balance")
+                )
+                line.price_subtotal_unbilled = line.price_unit - abs(
+                    line.price_subtotal_billed
+                )
```

Comment:
The logic doesn't look correct.  It should be like this if I'm not mistaken.  Please check and confirm.  Note that the sum of price_subtotal_unbilled is supposed to be the amount they should expect to pay after the current AP balance is cleared.

```suggestion
                line.price_subtotal_billed = 0.0
                line.price_subtotal_unbilled =  sum(
                    line.invoice_lines.filtered(
                        lambda l: l.move_id.state != "cancel"
                    ).mapped("balance")
                ) * -1
```

Reply:
I checked and I think your suggestion is correct.

Reply:
@AungKoKoLin1997 How about this, then?
```.py
                line.price_subtotal_billed =  sum(
                    line.invoice_lines.filtered(
                        lambda l: l.move_id.state != "cancel"
                    ).mapped("balance")
                )
                line.price_subtotal_unbilled = line.price_subtotal_billed * -1
```